### ISSUE TITLE
Question about tf.image.convert_image_dtype parameter type?

### ISSUE THREAD

**Issue by triumph-wangyuyang** [tensorflow/tensorflow#58699]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Documentation Bug

### Source

source

### Tensorflow Version

TF 2.10

### Custom Code

Yes

### OS Platform and Distribution

windows 11

### Python version

3.8.15

### CUDA/cuDNN version

CUDA: 11.3    cuDNN 8.6.0

### GPU model and memory

RTX3060

### Current Behaviour?

```shell
tf.image.convert_image_dtype(image, dtype, saturate=False, name=None), for image and dtype parameters can be uint8, uint16, uint32, uint64, int8, int16, int32, int64, float16, float32, float64, bfloat16. Then I set the image to complex64 and found it to work, so I don't know if the documentation is adequate.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
    image = tf.constant([[[254 + 2j]], [[83]], [[72]]], dtype=tf.complex64)
    dtype = tf.float64
    out = tf.image.convert_image_dtype(image, dtype)
    print(out)
```

### Relevant log output

```shell
result :
tf.Tensor(
  [[[inf]]

 [[inf]]

 [[inf]]], shape=(3, 1, 1), dtype=float64)
```
</details>

**Comment by triumph-wangyuyang** [comment]
The types of images in the tf.image.rgb_to_hsv operator must be half, bfloat16, float32, float64, and then I use float16, which can also run successfully.

code:
    import tensorflow as tf
    with tf.device('/CPU'):
      images = tf.random.uniform([1, 1, 3], dtype=tf.float16)
      out = tf.image.rgb_to_hsv(images)
      print(out)

**Comment by triumph-wangyuyang** [comment]
Although the x parameter type of tf.keras.activations.exponential(x) in the TensorFlow2.10 document is not clearly stated, when I deliberately use the wrong type int64 to test, I found that the type of x needs to be bfloat16, half, float, double, complex64, complex128 (obtained from the exception thrown by the test code below), which does not include float16, so I then used float16 to test and found that it was successful, but I don't understand why this situation exists.

code 1:

    import tensorflow as tf
    x = tf.random.uniform([5], minval=-256, maxval=257, dtype=tf.int64)
    out = tf.keras.activations.exponential(x)
    print(out)

result : Value for attr 'T' of int64 is not in the list of allowed values: bfloat16, half, float, double, complex64, complex128


code 2:

    import tensorflow as tf
    x = tf.random.uniform([5], minval=-256, maxval=257, dtype=tf.float16)
    out = tf.keras.activations.exponential(x)
    print(out)

result : tf.Tensor([inf  0. inf  0. inf], shape=(5,), dtype=float16)

**Comment by aadya940** [comment]
I think that an exception must be raised here.  if not image.dtype.complex64  must be added in the convert_image_dtype method

In case of your code, it is written in the docs that when casting from imaginary to real, tensorflow only returns the real part. Hence, no error was thrown.

**Comment by tiruk007** [comment]
@SuryanarayanaY 
I was able to reproduce the issue on Colab using Tf v2.11. Please find the gist [here](https://colab.research.google.com/gist/tiruk007/fcaf02ffba54de26fdd17bb12ebc4cac/58699.ipynb) for reference .

Thank you!

**Comment by SuryanarayanaY** [comment]
Hi @triumph-wangyuyang ,

Thanks for highlighting the issue. I also validated the `dtype `argument to `complex64` & `complex128` and it is raising Attribute Error as per attached [gist](https://colab.research.google.com/gist/SuryanarayanaY/4615b5d2b3b26ccfcb006915e411dc05/58699-r1.ipynb). We need to consider changing the documentation to include `tf.complex64` and `tf.complex128` as supported `dtypes` for image argument. 

Thankyou!

**Comment by oawxkw** [comment]
I also observed the following API aliases can cause the same issue in older versions of tensorflow.
Users should be cautious when using them on both CPU and GPU up to tensorflow 2.12.0 (v2.12.0-rc1-12-g0db597d0d75).

- `tf.image.convert_image_dtype`, `tf.compat.v1.image.convert_image_dtype`

It seems to be fixed in tensorflow 2.13.0 (v2.13.0-rc2-7-g1cb1a030a62) and later versions.
